### DISCUSSION TITLE
Module 'sqlalchemy' has no attribute 'dialects' - but it does

### DISCUSSION THREAD

**Discussion by DrWhen** [sqlalchemy/sqlalchemy#11358] [Usage Questions] [answered]
In one environment, python v3.12, but not in two other python v3.12 environments, I get the error: AttributeError: module 'sqlalchemy' has no attribute 'dialects'. 

The offending line seems to be: 
  File ~\Anaconda3\envs\py12\Lib\site-packages\sqlalchemy\dialects\__init__.py:47 in _auto_fn
    module = __import__("sqlalchemy.dialects.%s" % (dialect,)).dialects

To my surprise, without error, I can: import from sqlalchemy.dialects import postgresql; postgresql.UUID()

I have tried: conda update sqlalchemy and conda install conda-forge sqlalchemy

**Comment by CaselIT** [comment]
Hi,

How are you getting that error? Not really clear what you are trying to do here

**Reply by DrWhen** [reply]
I'm getting it when I build my engine.
dyBase_string = "DRIVER={ODBC Driver 17 for SQL Server}; SERVER=xxxxx; DATABASE=xxxx; UId=xxxxx; PWD=xxxxx"
dyBase_url = URL.create("mssql+pyodbc", 
                            query={"odbc_connect": dyBase_string})
dyBaseEngine = create_engine(dyBase_url, echo=False)

**Reply by DrWhen** [reply, marked as answer]
I've re-started the kernel, and it is now working.

**Comment by auderson** [comment]
Same issue here, can't figure out how to reproduce, but it does happen quite often.  Neverthelesss a restart will fix this.

**Reply by CaselIT** [reply]
Seems retated to your environment. Not really something that can be solved on the library side
